dbt_subprojects/daily_spellbook/models/_projects/gmx/event/arbitrum/gmx_v2_arbitrum_OrderCreated.sql

```diff
@@ -0,0 +1,279 @@
+        MAX(CASE WHEN key_name = 'initialCollateralDeltaAmount' THEN value END) AS initial_collateral_delta_amount,
+        MAX(CASE WHEN key_name = 'triggerPrice' THEN value END) AS trigger_price,
+        MAX(CASE WHEN key_name = 'acceptablePrice' THEN value END) AS acceptable_price,
+        MAX(CASE WHEN key_name = 'executionFee' THEN value END) AS execution_fee,
+        MAX(CASE WHEN key_name = 'callbackGasLimit' THEN value END) AS callback_gas_limit,
+        MAX(CASE WHEN key_name = 'minOutputAmount' THEN value END) AS min_output_amount,
+    
+        MAX(CASE WHEN key_name = 'updatedAtBlock' THEN value END) AS updated_at_block,
+        MAX(CASE WHEN key_name = 'updatedAtTime' THEN value END) AS updated_at_time,
+    
+        MAX(CASE WHEN key_name = 'isLong' THEN value END) AS is_long,
+        MAX(CASE WHEN key_name = 'shouldUnwrapNativeToken' THEN value END) AS should_unwrap_native_token,
+        MAX(CASE WHEN key_name = 'isFrozen' THEN value END) AS is_frozen,
+    
+        MAX(CASE WHEN key_name = 'key' THEN value END) AS key
+    FROM
+        combined
+    GROUP BY tx_hash, index
+)
+
+-- full data 
+, event_data AS (
+    SELECT 
+        blockchain,
+        block_time,
+        block_number,
+        ED.tx_hash,
+        ED.index,
+        contract_address,
+        event_name,
+        msg_sender,
+        topic1, 
+        topic2,
+        
+        account,
+        receiver,
+        callback_contract,
+        ui_fee_receiver,
+        market,
+        TRY_CAST(initial_collateral_token AS VARCHAR) AS initial_collateral_token,
+        
+        swap_path,
+        TRY_CAST(order_type AS INTEGER) AS order_type,
+        TRY_CAST(decrease_position_swap_type AS INTEGER) AS decrease_position_swap_type,
+        TRY_CAST(size_delta_usd AS DECIMAL) AS size_delta_usd,
+        TRY_CAST(initial_collateral_delta_amount AS DECIMAL) AS initial_collateral_delta_amount,
+        TRY_CAST(trigger_price AS DECIMAL) AS trigger_price,
+        TRY_CAST(acceptable_price AS DECIMAL) AS acceptable_price,
+        TRY_CAST(execution_fee AS DECIMAL) AS execution_fee,
+        TRY_CAST(callback_gas_limit AS DECIMAL) AS callback_gas_limit,
+        TRY_CAST(min_output_amount AS DECIMAL) AS min_output_amount,
+        updated_at_block,
+        TRY_CAST(updated_at_time AS DECIMAL) AS updated_at_time,
+        TRY_CAST(is_long AS BOOLEAN) AS is_long,
+        TRY_CAST(should_unwrap_native_token AS BOOLEAN) AS should_unwrap_native_token,
+        TRY_CAST(is_frozen AS BOOLEAN) AS is_frozen,
+        key
+        
+    FROM evt_data AS ED
+    LEFT JOIN evt_data_parsed AS EDP
+        ON ED.tx_hash = EDP.tx_hash
+            AND ED.index = EDP.index
+)
+
+-- Filter relevant tokens
+, collateral_tokens_data AS (
+    SELECT 
+        contract_address AS collateral_token, 
+        decimals AS collateral_token_decimals
+    FROM 
+        {{ source("ai_data_master", "result_token_information_from_arbitrum_api", database="dune") }}
+)
```

Comment:
my thought was instead of a CTE here, we would create a view as it's own spellbook model on top of this source. then read the spellbook view in this CTE here.

something like a new sql model file in: `../gmx/tokens/<chain>/`
where it's a simple select * from mat view
and in model config block (like line 5 in model here) would be `materialized = view` rather than table